File: Edward Poot/src/ql/ast/expr/binaryoperatorexpr/sub.go
package binaryoperatorexpr

import "ql/ast/expr"

type Sub struct {
	BinaryOperator
}

func NewSub(lhs expr.Expr, rhs expr.Expr) Sub {
	return Sub{BinaryOperator{Lhs: lhs, Rhs: rhs}}
}

func (s Sub) Eval(sy interface{}) interface{} {
	return s.Lhs.Eval(sy).(int) - s.Rhs.Eval(sy).(int)
}

File: NickLodewijks/src/org/uva/ql/ast/expr/Sub.java
package org.uva.ql.ast.expr;

import org.antlr.v4.runtime.ParserRuleContext;
import org.uva.ql.ast.ASTNodeVisitor;

public class Sub extends BinaryExpr {

	public Sub(ParserRuleContext context, Expr lhs, Expr rhs) {
		super(context, lhs, rhs);
	}

	@Override
	public <T, U> T accept(ASTNodeVisitor<T, U> visitor, U context) {
		return visitor.visit(this, context);
	}
}
